platform/podjs/src/browserPod.ts

```diff
@@ -15,68 +15,127 @@ import * as zip from "@zip.js/zip.js";
+        request.onerror = () => reject(request.error);
+    });
+}
+
+class IDBPolyIn implements PolyIn {
+    private async getStoredQuads(): Promise<RDF.Quad[]> {
+        const db = await openDatabase();
+        return new Promise((resolve, reject) => {
+            const tx = db.transaction(["poly-in"], "readonly");
+            const request = tx.objectStore("poly-in").getAll();
+            request.onsuccess = () => resolve(request.result);
+            tx.onerror = tx.onabort = () => reject(tx.error);
         });
     }
 
-    async add(...quads: RDF.Quad[]): Promise<void> {
-        this.store.push(...quads);
-        localStorage.setItem(
-            LocalStoragePolyIn.storageKey,
-            JSON.stringify(this.store)
+    async match(matcher: Partial<Matcher>): Promise<RDF.Quad[]> {
+        const storedQuads = await this.getStoredQuads();
+        return storedQuads.filter(
+            (quad) =>
+                (!matcher.subject || quad.subject.equals(matcher.subject)) &&
+                (!matcher.object || quad.object.equals(matcher.object)) &&
+                (!matcher.predicate || quad.predicate.equals(matcher.predicate))
         );
     }
 
+    select(matcher: Partial<Matcher>): Promise<RDF.Quad[]> {
+        return this.match(matcher);
+    }
+
+    async add(...quads: RDF.Quad[]): Promise<void> {
+        const db = await openDatabase();
+        return new Promise((resolve, reject) => {
+            const tx = db.transaction(["poly-in"], "readwrite");
+            const store = tx.objectStore("poly-in");
+            for (const quad of quads) store.add(quad);
+            tx.oncomplete = () => resolve();
+            tx.onerror = tx.onabort = () => reject(tx.error);
+        });
+    }
+
     async delete(...quads: RDF.Quad[]): Promise<void> {
-        quads.forEach((quad) => {
-            delete this.store[this.store.indexOf(quad)];
+        const db = await openDatabase();
+        return new Promise((resolve, reject) => {
+            const tx = db.transaction(["poly-in"], "readwrite");
+            const store = tx.objectStore("poly-in");
+            const request = store.openCursor();
+
+            request.onsuccess = () => {
+                const cursor = request.result;
+                if (cursor) {
+                    if (quads.some((quad) => quad.equals(cursor.value)))
+                        store.delete(cursor.key);
+                    cursor.continue();
+                }
+            };
+
+            tx.oncomplete = () => resolve();
+            tx.onerror = tx.onabort = () => reject(tx.error);
         });
-        localStorage.setItem(
-            LocalStoragePolyIn.storageKey,
-            JSON.stringify(this.store)
-        );
     }
 
     async has(...quads: RDF.Quad[]): Promise<boolean> {
-        throw `Called with ${quads}, not implemented: «has»`;
+        const storedQuads = await this.getStoredQuads();
+        return quads.some((a) => storedQuads.some((b) => a.equals(b)));
```

Comment:
I went ahead implementing this method as it only took two lines of code after the other changes in here. But from the documentations in the interface it is not clear whether it's supposed to return `true` if *some* or *all* of the given quads can be found. I went ahead implementing the former, but it would be great if someone could confirm that this is the correct behavior?

Reply:
The reference implementation is what we have https://github.com/polypoly-eu/polyPod/blob/ff029b592f380ba0046ba7e16c3c5e6d986e5143/platform/feature-api/api/pod-api/src/default.ts. Other implementations are actually based on this one.

Reply:
Good point - this one wasn't implemented because it's not yet used in production, but I also put in a one liner as part of #815 - because it seems like something we should support and test even if we don't use it in any feature right now.

As for what it _should_ do - yeah, right now `some` is apparently the intended behaviour. But when it comes down to it, we should just follow the actual specs instead, and only accept a single parameter, see: https://rdf.js.org/dataset-spec/

But that's for a separate PR I think.

Reply:
Good point - this one wasn't implemented because it's not yet used in production, but I also put in a one liner as part of #815 - because it seems like something we should support and test even if we don't use it in any feature right now.

As for what it _should_ do - yeah, right now `some` is apparently the intended behaviour. But when it comes down to it, we should just follow the actual specs instead, and only accept a single parameter, see: https://rdf.js.org/dataset-spec/

But that's for a separate PR I think.